### ISSUE TITLE
cmd: templ --watch does not close port on SIGINT

### ISSUE THREAD

**Issue by Murderlon** [a-h/templ#487]
Hi, I love the effort in bringing a watcher to `templ`! Unfortunately, the port of my server is never closed. 

I'm running a practically barebones [`chi`](https://github.com/go-chi/chi/) server, nothing special going on there.

- macOS Sonoma 14.2.1
- go version go1.21.5 darwin/arm64
- templ v0.2.543

```
λ templ generate --watch --proxy="http://localhost:3000" --cmd="go run ."                   

Processing path: /Users/merlijnvos/code/the-boring-stack
Generating dev code: /Users/merlijnvos/code/the-boring-stack
(✓) Generated code for "/Users/merlijnvos/code/the-boring-stack/views/layout.templ" in 1.523417ms
(✓) Generated code for "/Users/merlijnvos/code/the-boring-stack/views/index.templ" in 1.572042ms
(✓) Generated code for 2 templates with 0 errors in 2.030583ms
Executing command: go run .
Proxying from http://127.0.0.1:7331 to target: http://localhost:3000
Opening URL: http://localhost:3000
2024-02-02 15:04:06 INF starting server port=:3000
^CStopping...
Generating production code: /Users/merlijnvos/code/the-boring-stack
(!) Deleted file "/Users/merlijnvos/code/the-boring-stack/views/index_templ.txt"
(!) Deleted file "/Users/merlijnvos/code/the-boring-stack/views/layout_templ.txt"
(✓) Generated code for "/Users/merlijnvos/code/the-boring-stack/views/index.templ" in 2.432458ms
(✓) Generated code for "/Users/merlijnvos/code/the-boring-stack/views/layout.templ" in 2.350875ms
(✓) Generated code for 2 templates with 0 errors in 4.017542ms
Executing command: go run .

λ 2024-02-02 15:04:17 INF starting server port=:3000                                                                 
```
☝️ notice that the program stopped but it logged a line where my cursor is

Running it again:

```
λ templ generate --watch --proxy="http://localhost:3000" --cmd="go run ."                                        

Processing path: /Users/merlijnvos/code/the-boring-stack
Generating dev code: /Users/merlijnvos/code/the-boring-stack
(✓) Generated code for "/Users/merlijnvos/code/the-boring-stack/views/layout.templ" in 1.751917ms
(✓) Generated code for "/Users/merlijnvos/code/the-boring-stack/views/index.templ" in 1.754292ms
(✓) Generated code for 2 templates with 0 errors in 2.152833ms
Executing command: go run .
Proxying from http://127.0.0.1:7331 to target: http://localhost:3000
Opening URL: http://localhost:3000
2024-02-02 15:05:25 INF starting server port=:3000
listen tcp :3000: bind: address already in use
exit status 1
```

**Comment by a-h** [comment]
Would you mind giving the main branch a try please? I completely rewrote the whole of the `generate --watch` command in https://github.com/a-h/templ/pull/470

**Comment by Huzaifa-MS** [comment]
Coming from #493 
I tested on wsl on windows
using go: upgraded github.com/a-h/templ v0.2.543 => v0.2.544-0.20240205222607-1e6a80f83c3a
with command: /home/huz/go/bin/templ generate --watch --proxy=http://127.0.0.1:8080 --proxyport=8081 --cmd="go run ."

TLDR: Does not work. Spawns a new cmd after closing templ. claimed evidence is changed id of go process after ctrl+z

It seems that the templ process on ctrl+c is closing itself and the go process it started however after that it create a new go process? I say this based on the fact a the pgid and id of the go process has changed after I pressed ctrl+c to terminate.

![proof-templ](https://github.com/a-h/templ/assets/43180782/265bff46-f19d-4de4-9ed4-157968a61a16)

I was trying to run templ using custom hotreload go file and ran into this issue on linux. for some reason I can easily kill the command in windows using my custom hotreload

**Comment by a-h** [comment]
Hi @Huzaifa-MS, from the logs you posted, it looks like you're not using the latest version of the CLI on the main branch.

Would you be able to test it with the main branch instead please? I've rewritten the whole of the --watch mode code from scratch since the version you're using.

**Comment by Huzaifa-MS** [comment]
I will try. Sorry I thought doing a go get to main would get the latest

**Comment by Huzaifa-MS** [comment]
It is working correctly now. Great work! :)

Steps: 
*New Templ Repo needs slog to compile meaning it needs a Go version which has this. My previous version 1.20 did not have it.
1. git clone https://github.com/a-h/templ.git
2. cd templ/cmd/templ
3. go build
4. go install.
5. templ version #should be 2.560 or greater
    *go get -u no longer works as in installing package from what I understand in newer go versions

Here is the screenshot:
[
![proof-templ-2](https://github.com/a-h/templ/assets/43180782/3e7d1ea6-ace1-4848-bd6f-6de27e41645e)
](url)


Also how would I fix the generator warning? or will updating to next release fix itself

**Comment by Huzaifa-MS** [comment]
Sorry This is unrelated but is there any way to get output to show after using proxy flag?

![templ proxy command block cli 2](https://github.com/a-h/templ/assets/43180782/8a85a258-f871-4327-904b-54a2dac38b2c)

It's blocking any logs from the changes I make in .templ files from showing up. I believe this is due to XDISPLAY in WSL

**Comment by a-h** [comment]
It should fix itself in the next release.

With Go, when you do `go get -u github.com/a-h/templ` it installs the latest tag, based on a semantic version, e.g. v0.0.450.

If you run `go get -u github.com/a-h/templ@<commithash>` it installs that specific version.

You don't need Go 1.21 installed to run the CLI, if you use a binary that we release, but obviously, if you build from source, you'll need the later Go version.

The strategy we have is to use the latest Go version to build the CLI and produce binaries, but the runtime supports the current Go version - 2 releases. Go 1.22 was released today, so we're now free to start using Go 1.20 features in the runtime.

**Comment by Huzaifa-MS** [comment]
Thank you for helping me understand this better : )

**Comment by a-h** [comment]
@Murderlon - the latest release of templ should have fixed this issue. If not, please feel free to reopen, and I'll start investigation again, but here's a demo of it working fine, with the example `chi` project in the templ repo.

https://github.com/a-h/templ/assets/1029947/7888fb2f-232d-420d-9b75-ceaba996b4c0

**Comment by Murderlon** [comment]
Sorry I didn't get around to testing this earlier myself. Thanks for the fix!
